graylog2-server/src/main/java/org/graylog2/decorators/DecoratorProcessorImpl.java

```diff
@@ -87,8 +88,14 @@ public SearchResponse decorate(SearchResponse searchResponse, Optional<String> s
 
     private Set<String> extractFields(List<ResultMessageSummary> messages) {
         return messages.stream()
-            .map(message -> message.message().keySet())
-            .reduce(new HashSet<>(), (set1, set2) -> { set1.addAll(set2); return set1; });
+                .map(message -> message.message().keySet())
+                .reduce(new HashSet<>(), (set1, set2) -> {
+                    set1.addAll(set2);
+                    return set1;
+                })
+                .stream()
+                .filter(field -> !Message.RESERVED_FIELDS.contains(field))
+                .collect(Collectors.toSet());
```

Comment:
This could be simplified to:

`return messages.stream()
                .flatMap(message -> message.message().keySet().stream())
                .filter(field -> !Message.RESERVED_FIELDS.contains(field))
                .collect(Collectors.toSet());`

which is easier to read imo.